### ISSUE TITLE
[Spike] - Identify Appointment Types being Displayed in VAOS

### ISSUE THREAD

**Issue by JRRoof** [department-of-veterans-affairs/va.gov-team#52886]
## Background/Request
It is believed that Compensation & Pension (C&P) appointments are now being displayed in VAOS after the move to the new VAOS service.  However, we have no evidence that this is happening.  This request is to create logging to capture the different appointment types being displayed in VAOS to veterans.

### Goal
The goal is to determine if C&P appointments are currently being displayed in VAOS.  Secondary to that goal would be to capture any other appointment types that are also being displayed.

### Requirements to Consider
Not looking to capture details of the C&P Appointments.  Ideally, a count of the number of different appointment types would be useful, but not necessary. The main goal is to identify the distinct types of appointments being displayed in VAOS.

### **Technical Notes**
To determine the appointment type look at `Appointment.serviceCategory`

A Compensation & Pension appointment will look like this:
```
serviceCategory = [
  {
    "coding": [
      {
      "system": "http://www.va.gov/Terminology/VistADefinedTerms/409_1",
      "code": "COMPENSATION & PENSION"
      }
    ]
]
```
This is the list of national values that will appear for the code:

```
COMPENSATION & PENSION
CLASS II DENTAL
ORGAN DONORS
EMPLOYEE
PRIMA FACIA
RESEARCH
COLLATERAL OF VET.
SHARING AGREEMENT
REGULAR
COMPUTER GENERATED
SERVICE CONNECTED
```
If a site has defined a custom appointment type _it's possible_ you might see other values.  An appointment type of `REGULAR` is a health appointment.

To test this in SQA you need an appointment created that has that appt type, and the clinic has to be returned by the CDW service with `displayAppointment` set to true.

For testing, create an appointment in an existing clinic in SQA (that is technically not a C&P clinic).  Per Marcy, we don't have any C&P clinics configured in Staging.  However, she can make an appointment for you in a clinic and set the appointment type to C&P.  If she needs to create C&P Clinics for testing then that takes time, because she needs to know stop codes and then log a ticket to get created in Mock CDW.  Use `Judy Morrison` as the test user.

---

## Tasks
- Answer the following questions:
  - Q  Are C&P appointments being displayed in VAOS?
  - Q  What other appointment types are also being displayed in VAOS?

## Time Box
6 hours

## Definition of Done
- Tasks are complete
- Research and recommendations are documented in this ticket

---
## How to configure this issue
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)

**Comment by cferris32** [comment]
Datadog findings from the first ~24 hours of VAOS service category monitoring:

From about 13k logs, we see that a vast majority of them are health appointments (service category == REGULAR) with a variety of other categories appearing in much smaller numbers. 

Q1: Are compensation & pension appointments being displayed in VAOS?
A1: Yes. While the amount is small, we've seen a few C&P appointments roll through our app's activity:
![Screenshot 2023-02-28 at 3.47.07 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/be733caa-ec34-44a5-b62a-7b428a9eb709)


Q2: What other appointment types are being displayed in VAOS?
A2: From our current data set, most service category values are appearing in VAOS, albeit in relatively small amounts still. These include "CLASS II DENTAL", "SERVICE CONNECTED", "COLLATERAL OF VET". "SHARING AGREEMENT", "PRIMA FACIA", "RESEARCH", and "EMPLOYEE". The only ones not seen at the time of this writing are "ORGAN DONORS" and "COMPUTER GENERATED" which could be due to a number of reasons including possibly not enough data gathered yet. Regardless, it's clear from this initial dive that we see many kinds of appointment service categories being handled by VAOS. Below are some additional screenshots showing the total counts of different log groups that were examined, along with proportional counts of the service categories that appeared in each group.

Total log counts:
![Screenshot 2023-02-28 at 3.13.54 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/bd1040a9-e539-4f25-8421-cfeb09130d56)
Count of "SERVICE CONNECTED" within its group
![Screenshot 2023-02-28 at 3.18.09 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/a355913a-f6d7-4fe4-8c05-3d1c3938d831)
Additional "SERVICE CONNECTED" counts from another group
![Screenshot 2023-02-28 at 3.18.36 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/d63b73c4-14a8-4015-b81d-0007697e70bf)
Counts of "RESEARCH" and "EMPLOYEE" within their group
![Screenshot 2023-02-28 at 3.18.21 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/19dae493-f31e-4659-9968-00720bbffa93)
Counts of "SHARING AGREEMENT" and "PRIMA FACIA" within their group
![Screenshot 2023-02-28 at 3.18.48 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/6076abfc-76b5-4d1d-9019-116191be89d6)
"COLLATERAL OF VET"
![Screenshot 2023-02-28 at 4.10.16 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/2d4967da-c445-4367-95c9-d760842bc5b4)
Lastly, a single "CLASS II DENTAL" entry was found
![Screenshot 2023-02-28 at 3.19.23 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/064257f0-712a-425d-8144-21fc9fcf9606)

In summary, while the amounts are relatively small compared to the health appointments, there's evidence of VAOS handling appointments of many service category types expanding into non-health appointment scheduling.
